File: aula1/exercicio3.py
def Primo (n):
  j = 2
  while ((j < n) & ((n % j) != 0) ):
    j=j+1
  if (j == n):
    print(f"{n} e primo")
  else:
    print(f"{n} nao e primo")

Minimo = 5
Maximo = 3001
entradas = range(2,Maximo)
tempos = []
import timeit, functools
for n in entradas:
  t = timeit.Timer(functools.partial(Primo, n) )
  if (n >= Minimo):
    tempos.append(t.timeit(5))  # Armazena o tempo

# Criar o gráfico
import matplotlib.pyplot as plt
entradas = range(Minimo,Maximo)
plt.plot(entradas, tempos)
plt.title("Tempo de execução da função 'Primos' em função de n")
plt.xlabel("Valor de n")
plt.ylabel("Tempo de execução (segundos)")
plt.grid(True)
plt.show()

File: aula1/exercicio3.cpp
//
// Created by 60659211220 on 17/11/2024.
//
#include <stdio.h>
#include <stdlib.h>

void Primo(int n)
{
  int j = 2;
  while ((j < n) && ((n % j) != 0) ) {
     j++;
  }
  if (j == n) {
    printf("%d e primo", n);
  }
  else{
    printf("%d nao e primo", n);
  }
   
}

int main() {
  Primo(7);
  return 0;
}